### PR TITLE
chore(react-native): e2e tests creation and enhancement of flow

### PR DESCRIPTION
**Pull request by vanGalilea**
## Goal
- Move all E2E tests to DF app
- Tailor test flow to DF app
- Enhance flow with log in/out and partic. info

## Screen Recording

https://github.com/GetStream/stream-video-js/assets/25864161/71055363-cec6-44c8-a70e-5c137d8e4d14

https://github.com/GetStream/stream-video-js/assets/25864161/abd101cc-b43e-46e6-a13c-dbb376a45bd7

## Would you like to run the tests locally?
- Follow the steps in [README.md](https://github.com/GetStream/stream-video-js/pull/640/files#diff-1dcda0c8086ff25742f32b61658d1a2fa7f564daac8c99d768f5f56529adbb7cR23)

### CODE REVIEW

File: packages/react-native-dogfood/src/components/NavigationHeader.tsx [hunk 1]
```diff
@@ -4,30 +4,23 @@ import {
...
+import { SafeAreaView } from 'react-native-safe-area-context';
 
 const styles = StyleSheet.create({
   header: {
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    alignItems: 'center',
-    paddingHorizontal: 10,
-    paddingTop: 20,
-    paddingBottom: 20,
```
Comment (vanGalilea) [inline packages/react-native-dogfood/src/components/NavigationHeader.tsx:20]: Removed unneeded styling I stumbled upon

File: packages/react-native-dogfood/src/components/NavigationHeader.tsx [hunk 2]
```diff
@@ -65,8 +58,11 @@ export const NavigationHeader = () => {
   };
 
   return (
-    <View style={styles.header}>
-      <Pressable onPress={logoutHandler}>
+    <SafeAreaView style={styles.header}>
```
Comment (vanGalilea) [inline packages/react-native-dogfood/src/components/NavigationHeader.tsx:61]: replace with normal view

File: packages/react-native-dogfood/src/screens/Meeting/JoinMeetingScreen.tsx [hunk 3]
```diff
@@ -42,7 +43,7 @@ const JoinMeetingScreen = (props: JoinMeetingScreenProps) => {
   );
 
   return (
-    <View style={styles.container}>
+    <SafeAreaView style={styles.container}>
```
Comment (vanGalilea) [inline packages/react-native-dogfood/src/screens/Meeting/JoinMeetingScreen.tsx:46]: Was needed to be able to click on the avatar in order to logout.
The avatar was partially under the status bar which made it unclickable

File: packages/react-native-sdk/src/components/CallParticipantsInfoView.tsx [hunk 4]
```diff
@@ -180,11 +179,12 @@ export const CallParticipantsInfoView = ({
           <View style={styles.header}>
             <View style={styles.leftHeaderElement} />
             <Text style={styles.headerText}>
-              Participants ({participantCount})
+              Participants ({participants.length})
```
Comment (vanGalilea) [inline packages/react-native-sdk/src/components/CallParticipantsInfoView.tsx:182]: Partic. count originating from `useParticipantCount` has a certain lag, due to BE.
We want to have the most up-to-date count of partic. therefore we use `participants.length`

**Review by vanGalilea — COMMENTED** [review]
Self reviewed ✅

**Review by khushal87 — APPROVED** [review]

**Pull request merged**